pkg/plugin/v3/init.go

```diff
@@ -167,3 +173,26 @@ func (p *initPlugin) PostScaffold() error {
 	fmt.Printf("Next: define a resource with:\n$ %s create api\n", p.commandName)
 	return nil
 }
+
+// checkDir will return error if the current directory has files which are
+// not the go.mod and/or starts with the prefix (.) such as .gitignore.
+// Note that, it is expected that the directory to scaffold the project is cleaned.
+// Otherwise, it might face issues to do the scaffold. The go.mod is allowed because user might run
+// go mod init before use the plugin it for not be required inform
+// the go module via the repo --flag.
+func checkDir() error {
+	err := filepath.Walk(".",
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.Name() != "go.mod" && !strings.HasPrefix(info.Name(), ".") {
```

Comment:
The condition here should be OR
```suggestion
			if info.Name() != "go.mod" || !strings.HasPrefix(info.Name(), ".") {
```